### PR TITLE
INTYGFV-15844: Implementation to ensure that messages cannot be sent …

### PR DESCRIPTION
**Pull request by mikaelaromild1**
…unless certificate is already sent.

### CODE REVIEW

File: web/src/test/java/se/inera/intyg/intygstjanst/web/integration/validator/SendMessageToRecipientValidatorTest.java [hunk 1]
```diff
@@ -221,6 +225,31 @@ public void normalizeCrnBeforeCompareTest() throws Exception {
         assertTrue(validationErrors.isEmpty());
     }
 
+    @Test
+    public void shouldContainValidationErrorIfCertificateIsNotSent() throws InvalidCertificateException {
+        when(certificateService.getCertificateForCare(INTYG_ID)).thenReturn(buildCertificate(PATIENT_CRN, CertificateState.UNHANDLED));
+        List<String> validationErrors = validator.validate(buildOkMessage(PATIENT_CRN));
```
Comment (ChrilleSigerhed) [inline web/src/test/java/se/inera/intyg/intygstjanst/web/integration/validator/SendMessageToRecipientValidatorTest.java:231]: Maybe change this to ```final var``` intead of ```List<String>``` to match the rest of the implementation?

File: web/src/test/java/se/inera/intyg/intygstjanst/web/integration/validator/SendMessageToRecipientValidatorTest.java [hunk 2]
```diff
@@ -221,6 +225,31 @@ public void normalizeCrnBeforeCompareTest() throws Exception {
...
+
+        assertEquals(1, validationErrors.size());
+    }
+
+    @Test
+    public void shouldContainSpecificValidationErrorIfCertificateIsNotSent() throws InvalidCertificateException {
+        when(certificateService.getCertificateForCare(INTYG_ID)).thenReturn(buildCertificate(PATIENT_CRN, CertificateState.UNHANDLED));
+        List<String> validationErrors = validator.validate(buildOkMessage(PATIENT_CRN));
```
Comment (ChrilleSigerhed) [inline web/src/test/java/se/inera/intyg/intygstjanst/web/integration/validator/SendMessageToRecipientValidatorTest.java:239]: Maybe change this to ```final var``` intead of ```List<String>``` to match the rest of the implementation?

File: web/src/test/java/se/inera/intyg/intygstjanst/web/integration/validator/SendMessageToRecipientValidatorTest.java [hunk 3]
```diff
@@ -221,6 +225,31 @@ public void normalizeCrnBeforeCompareTest() throws Exception {
...
+        final var expectedResult = "Certificate is not sent to recipient";
+
+        assertTrue(validationErrors.contains(expectedResult));
+    }
+
+    @Test
+    public void shouldNotContainValidationErrorIfCertificateIsSent() throws InvalidCertificateException {
+        when(certificateService.getCertificateForCare(INTYG_ID)).thenReturn(buildCertificate(PATIENT_CRN, CertificateState.SENT));
+        List<String> validationErrors = validator.validate(buildOkMessage(PATIENT_CRN));
```
Comment (ChrilleSigerhed) [inline web/src/test/java/se/inera/intyg/intygstjanst/web/integration/validator/SendMessageToRecipientValidatorTest.java:248]: Maybe change this to ```final var``` intead of ```List<String>``` to match the rest of the implementation?

**Review by ChrilleSigerhed — CHANGES_REQUESTED** [review]
Well done, looks great! Added comments related to the type of validationsErrors in your tests, maybe we should change it to final var for uniformly purposes?

**Comment by mikaelaromild1** [comment]
Made changes according to our discussion.

**Review by ChrilleSigerhed — APPROVED** [review]
Well done!

**Pull request merged**